workers/ohsome_quality_analyst/indicators/last_edit/indicator.py

```diff
@@ -23,83 +25,79 @@ def __init__(
+        response = await ohsome_client.query(
             layer=self.layer,
             bpolys=json.dumps(self.bpolys),
         )
-        if query_results_contributions is None or query_results_totals is None:
-            return False
-        try:
-            self.edited_features = len(query_results_contributions["features"])
-        except KeyError:
-            # no feature has been edited in the time range
-            self.edited_features = 0
-
-        self.total_features = query_results_totals["result"][0]["value"]
-        if self.total_features != 0 and self.total_features is not None:
-            if self.edited_features > self.total_features:
-                # It can happen that features are counted which has been deleted since.
-                self.share_edited_features = 100
-            else:
-                self.share_edited_features = round(
-                    (self.edited_features / self.total_features) * 100
-                )
+        self.element_count = response["result"][0]["value"]
+
         return True
 
     def calculate(self) -> bool:
         logging.info(f"Calculation for indicator: {self.metadata.name}")
 
-        description = Template(self.metadata.result_description).substitute(
-            share=self.share_edited_features, layer_name=self.layer.name
-        )
-        if self.total_features == 0:
+        # It can be that features are counted, but have been deleted since.
+        if self.element_count == 0:
             self.result.label = "undefined"
             self.result.value = None
-            self.result.description = (
-                "Since the OHSOME query returned a count of 0 for this feature "
-                "a quality estimation can not be made for this filter"
-            )
+            self.result.description = "In the AOI no features are present."
             return False
-        elif self.share_edited_features >= self.threshold_yellow:
+
+        if self.contributions_latest_count > self.element_count:
+            self.ratio = 100
+        else:
+            self.ratio = round(
+                (self.contributions_latest_count / self.element_count) * 100
+            )
+
+        description = Template(self.metadata.result_description).substitute(
+            share=self.contributions_latest_count, layer_name=self.layer.name
+        )
+
+        if self.ratio >= self.threshold_yellow:
             self.result.label = "green"
             self.result.value = 1.0
             self.result.description = (
                 description + self.metadata.label_description["green"]
             )
-        elif self.share_edited_features >= self.threshold_red:
+        elif self.ratio >= self.threshold_red:
             self.result.label = "yellow"
             self.result.value = 0.5
             self.result.description = (
                 description + self.metadata.label_description["yellow"]
             )
-        else:
+        elif self.ratio < self.threshold_red:
```

Comment:
Not sure what the best programming behaviour is here? If any `elif` case is ever changed here, you wouldn't have covered cases. Maybe you can add a `else:` case with an exception that shouldn't ever occur?

Reply:
The ValueError exception will now be raised once the else branch is reached.